classes/helper.php

```diff
@@ -96,23 +149,35 @@ private static function regular_expression_search(string $search, string $table,
 
         $results = [];
 
-        $columns = $DB->get_columns($table);
+        $columns = self::get_columns($table, $column);
 
-        if ($column !== self::ALL_COLUMNS) {
-            // Only search the specified column.
-            $columns = array_filter($columns, function($col) use ($column) {
-                return $col->name == $column;
-            });
-        }
+        // Find Potential course field in the table.
+        $coursefield = self::find_course_field($table);
+
+        // Table alias.
+        $tablealias = 't';
 
         foreach ($columns as $column) {
             $columnname = $DB->get_manager()->generator->getEncQuoted($column->name);
 
-            $select = $columnname . ' ' . $DB->sql_regex() . ' :pattern ';
+            $select = "$tablealias." . $columnname . ' ' . $DB->sql_regex() . ' :pattern ';
             $params = ['pattern' => $search];
 
             if ($column->meta_type === 'X' || $column->meta_type === 'C') {
-                $records = $DB->get_records_select($table, $select, $params, '', '*', 0, $limit);
+
+                if (!empty($coursefield)) {
+                    $sql = "SELECT $tablealias.id,
+                                   $tablealias.$columnname,
+                                   $tablealias.$coursefield as courseid,
+                                   c.idnumber as courseidnumber
+                              FROM {".$table."} $tablealias
+                              JOIN {course} c ON c.id = $tablealias.$coursefield
```

Comment:
maybe left join this in case the field is nullable

Reply:
Hi @brendanheywood 

Thanks for your review.
I have added the left join.